### ISSUE TITLE
PDF Generated not showing data instead showing loading icon

### ISSUE THREAD

**Issue by Ramesh-Karnati** [IzakMarais/reporter#149]
PDF Generated not showing data instead showing loading icon. I have 10 panels in 1 Dashboard, When I generate PDF some of the panels are showing with loading icon instead of data.
![Error](https://user-images.githubusercontent.com/5021240/63015745-6be16a80-beaf-11e9-8d23-ee5491fabdc9.PNG)
![Original](https://user-images.githubusercontent.com/5021240/63015746-6be16a80-beaf-11e9-836e-18eda8144269.PNG)

**Comment by IzakMarais** [comment]
This is likely a performance problem on your time series back-end or your individual graph queries. 

Are there any error or warning messages in the reporter logs?

**Comment by Ramesh-Karnati** [comment]
No Errors. Can we have some sleep time between each downloading image

2019/08/14 19:03:19 Downloading image  2 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=2&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:19 Downloading image  10 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=10&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:19 Downloading image  8 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=8&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:19 Downloading image  4 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=4&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:19 Downloading image  6 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=6&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:23 Downloading image  12 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=12&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:23 Downloading image  18 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=18&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:23 Downloading image  14 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=14&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:23 Downloading image  16 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=16&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:23 Downloading image  22 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=22&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:26 Downloading image  20 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=20&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:26 Downloading image  26 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=26&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000
2019/08/14 19:03:26 Downloading image  24 http://localhost:3000/render/d-solo/OWSoG-gWz/_?from=now-5y&height=500&panelId=24&theme=light&to=now&var-Currency=Euro&var-packageID=I180268401&width=1000

**Comment by Ramesh-Karnati** [comment]
The issue is, the panel is not loading data by the time pdf is being generated, hence the loading icon is still coming

**Comment by IzakMarais** [comment]
> Can we have some sleep time between each downloading image

You can edit the source code to see if it solves your problem.

Reducing the number of workers [here](https://github.com/IzakMarais/reporter/blob/master/report/report.go#L139) from 5 to 1 will download the images in series in stead of in parallel. You can also add sleep after each call to `renderPNG()`.

**Comment by Ramesh-Karnati** [comment]
ok. I have reduced the workers from 5 to 1. Could you please guide where I can keep that sleep command

**Comment by Ramesh-Karnati** [comment]
Is that before line 146 in report.go?

**Comment by Ramesh-Karnati** [comment]
Also, is there any possible way that while doing http req call in line 139 in api.go , we can have some sleep time to get the response. I guess response.body is still loading by the time it was getting the response.

**Comment by IzakMarais** [comment]
> 
> 
> Is that before line 146 in report.go?

Yes, directly after the call to [renderPNG()](https://github.com/IzakMarais/reporter/blob/master/report/report.go#L146) should do what you want it to. 

> Also, is there any possible way that while doing http req call in line 139 in api.go , we can have some sleep time to get the response. I guess response.body is still loading by the time it was getting the response.

No,  the fact that you are seeing completely rendered images with no error messages implies there is nothing we can do on the caller side during the HTTP request to change the behaviour; Grafana should not have sent the response yet.

**Comment by brucexie** [comment]
Hi all, 
Ask a question, how to adjust the download image High size? because my Grafana dashboard Graph added more than 6 row table data, but the download and generate the image only show 6 row, from 7 row table data won't be show and download into PDF report.
